test/datadog/e2e_eks_test.go

```diff
@@ -0,0 +1,79 @@
+package datadog
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/infra"
+	"github.com/DataDog/helm-charts/test/common"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/eks"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	ddaPrefix = "dda-datadog"
+	dcaPrefix = "dda-datadog-cluster-agent"
+	ccPrefix  = "dda-datadog-clusterchecks"
+)
+
+func Test_E2E_AgentOnEKS(t *testing.T) {
+	// Create pulumi EKS stack
+	config := common.SetupConfig()
+	stackConfig := runner.ConfigMap{
+		"ddinfra:aws/eks/linuxBottlerocketNodeGroup": auto.ConfigValue{Value: "false"},
+		"ddinfra:aws/eks/windowsNodeGroup":           auto.ConfigValue{Value: "false"},
+		"pulumi:disable-default-providers":           auto.ConfigValue{Value: "[]"},
+		"aws:skipCredentialsValidation":              auto.ConfigValue{Value: "true"},
+		"aws:skipMetadataApiCheck":                   auto.ConfigValue{Value: "false"},
+	}
+	stackConfig.Merge(config)
+
+	_, stackOutput, err := infra.GetStackManager().GetStack(context.Background(), "eks-e2e", stackConfig, eks.Run, false)
+	defer common.TeardownSuite(PreserveStacks)
+
+	if stackOutput.Outputs["kubeconfig"].Value != nil {
+		kc := stackOutput.Outputs["kubeconfig"].Value.(map[string]interface{})
+
+		_, restConfig, clientSet, err := common.NewClientFromKubeconfig(kc)
+		require.NoError(t, err)
+
+		namespace := "datadog"
+		nodes, err := common.ListNodes(namespace, clientSet)
+		require.NoError(t, err)
+
+		pods, err := common.ListPods(namespace, clientSet)
+		ddaPods := 0
+		dcaPods := 0
+		ccPods := 0
+
+		for _, pod := range pods.Items {
```

Comment:
I think test would be more readable if this counting logic is moved to a separate function or to `common_e2e.go`. 
Alternatively one could utilize `metav1.ListOptions` to list pods for each deployment separately, passing something like this in the list call
```
v1.ListOptions{
   LabelSelector: "app=dda-datadog-cluster-agent"
}
```

Reply:
Agreed--I've refactored the logic into a separate `VerifyPods` function in `common_e2e.go`: https://github.com/DataDog/helm-charts/pull/1049/commits/9cdf72e5138e79712e66cf798ba371b232713e68